docs/DESIGN.md

```
@@ -0,0 +1,429 @@
+    C[TelemetryDecorator<br/>로그 / 메트릭 / waiter count]
+    D[HeartbeatDecorator<br/>60s+ 진행 중 entry 경고]
+    E[CapacityDecorator<br/>30+ waiter 거부]
+    F[DeadlineDecorator<br/>180s wall-clock 타임아웃]
+    G[InProcessSingleFlightCoordinator<br/>ConcurrentHashMap.computeIfAbsent]
+    H[ensureSessionImpl<br/>Playwright 영역]
+
+    A -->|depends on Port| B
+    B -.->|implements| C
+    C -->|inner| D
+    D -->|inner| E
+    E -->|inner| F
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)